spring-cloud-generator/scripts/generate-showcase.sh

```diff
@@ -1,111 +1,175 @@
     exit 1
   fi
 
   # Clone sdk-platform-java (with showcase library)
-  git clone https://github.com/googleapis/sdk-platform-java.git
-  cd sdk-platform-java && git checkout "v${GAPIC_GENERATOR_JAVA_VERSION}"
+  if [[ ! -d "./sdk-platform-java" ]]; then
+    git clone https://github.com/googleapis/sdk-platform-java.git
+  fi
+  pushd sdk-platform-java
+  git checkout "v${gapic_generator_java_version}"
+
+  # We will use the generation tools from library_generation
+  pushd library_generation/utils
+  source utilities.sh
+  popd #library_generation/utils
 
   # Install showcase client libraries locally
-  cd showcase && mvn clean install
-  GAPIC_SHOWCASE_CLIENT_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+  pushd showcase
+  # For local development, we cleanup any traces of previous runs
+  rm -rdf output
+  mvn clean install
+  gapic_showcase_client_version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+
+  pushd gapic-showcase
+  gapic_showcase_server_version=$(mvn help:evaluate -Dexpression=gapic-showcase.version -q -DforceStdout)
+  popd #showcase/gapic-showcase
+
+  output_folder=$(get_output_folder)
+  mkdir "${output_folder}"
+  pushd "${output_folder}"
+  protoc_version=$(get_protoc_version "${gapic_generator_java_version}")
+  os_architecture=$(detect_os_architecture)
+  download_protoc "${protoc_version}" "${os_architecture}"
+
+  # We now copy the spring-cloud-generator jar with dependencies
+  # into the output_folder the sdk-platform-java generation
+  # scripts work with.
+  spring_generator_jar_name="spring-cloud-generator-${project_version}-jar-with-dependencies.jar"
+  cp ~/.m2/repository/com/google/cloud/spring-cloud-generator/"${project_version}/${spring_generator_jar_name}" \
+    "${output_folder}"
+  chmod 555 ${output_folder}/*.jar
+
+  # We download gapic-showcase and prepare the protos in output_folder
+  sparse_clone https://github.com/googleapis/gapic-showcase.git "schema/google/showcase/v1beta1"
+  pushd gapic-showcase
+  git checkout "v${gapic_showcase_server_version}"
+  cp -r schema "${output_folder}"
+  popd #gapic-showcase
+
+  # We download googleapis and prepare the protos in output_folder
+  if [[ ! -d "./googleapis" ]]; then
+    sparse_clone https://github.com/googleapis/googleapis.git "google/iam/v1 google/cloud/location google/api google/longrunning google/rpc google/type"
+  fi
+  pushd googleapis
+  cp -r google "${output_folder}"
+  popd #googleapis
+
+  # Now we call protoc with a series of arguments we obtain from
+  # sdk-platform-java's utilities.sh and others that are hardcoded (and stable).
+  # Note that --java_gapic_spring_opt uses `get_gapic_opts` which will work
+  # since the BUILD rules take similar arguments
+
+  proto_path="schema/google/showcase/v1beta1"
+  proto_files=$(find "${proto_path}" -type f  -name "*.proto" | LC_COLLATE=C sort)
+  gapic_additional_protos="google/iam/v1/iam_policy.proto google/cloud/location/locations.proto"
```

Comment:
I don't think we need these additional protos, same thing for `rest_numeric_enums`, `transport`, `gapic_yaml`, `include_samples` etc. Can you try to remove them and see if the checks are still passing?

Reply:
@blakeli0 when removing the additional protos, the generated code has missing `iam` methods. For example:

```
diff -r /usr/local/google/home/diegomarquezp/Desktop/spring-cloud-gcp/spring-cloud-generator/scripts/../showcase/showcase-spring-starter-generated/src/main/java/com/google/showcase/v1beta1/spring/SequenceServiceSpringAutoConfiguration.java /usr/local/google/home/diegomarquezp/Desktop/spring-cloud-gcp/spring-cloud-generator/scripts/../showcase/showcase-spring-starter/src/main/java/com/google/showcase/v1beta1/spring/SequenceServiceSpringAutoConfiguration.java
98a99,101
>     if (this.clientProperties.getUseRest()) {
>       return SequenceServiceSettings.defaultHttpJsonTransportProviderBuilder().build();
>     }
123c126,134
<     SequenceServiceSettings.Builder clientSettingsBuilder = SequenceServiceSettings.newBuilder();
---
>     SequenceServiceSettings.Builder clientSettingsBuilder;
>     if (this.clientProperties.getUseRest()) {
>       clientSettingsBuilder = SequenceServiceSettings.newHttpJsonBuilder();
>       if (LOGGER.isTraceEnabled()) {
>         LOGGER.trace("Using REST (HTTP/JSON) transport.");
>       }
>     } else {
>       clientSettingsBuilder = SequenceServiceSettings.newBuilder();
>     }
186a198,224
>       RetrySettings listLocationsRetrySettings =
>           RetryUtil.updateRetrySettings(
>               clientSettingsBuilder.listLocationsSettings().getRetrySettings(), serviceRetry);
>       clientSettingsBuilder.listLocationsSettings().setRetrySettings(listLocationsRetrySettings);
> 
>       RetrySettings getLocationRetrySettings =
>           RetryUtil.updateRetrySettings(
>               clientSettingsBuilder.getLocationSettings().getRetrySettings(), serviceRetry);
>       clientSettingsBuilder.getLocationSettings().setRetrySettings(getLocationRetrySettings);
> 
>       RetrySettings setIamPolicyRetrySettings =
>           RetryUtil.updateRetrySettings(
>               clientSettingsBuilder.setIamPolicySettings().getRetrySettings(), serviceRetry);
>       clientSettingsBuilder.setIamPolicySettings().setRetrySettings(setIamPolicyRetrySettings);
> 
>       RetrySettings getIamPolicyRetrySettings =
>           RetryUtil.updateRetrySettings(
>               clientSettingsBuilder.getIamPolicySettings().getRetrySettings(), serviceRetry);
>       clientSettingsBuilder.getIamPolicySettings().setRetrySettings(getIamPolicyRetrySettings);
> 
>       RetrySettings testIamPermissionsRetrySettings =
>           RetryUtil.updateRetrySettings(
>               clientSettingsBuilder.testIamPermissionsSettings().getRetrySettings(), serviceRetry);
>       clientSettingsBuilder
>           .testIamPermissionsSettings()
>           .setRetrySettings(testIamPermissionsRetrySettings);
> 
254a293,346
>       }
>     }
>     Retry listLocationsRetry = clientProperties.getListLocationsRetry();
>     if (listLocationsRetry != null) {
>       RetrySettings listLocationsRetrySettings =
>           RetryUtil.updateRetrySettings(
>               clientSettingsBuilder.listLocationsSettings().getRetrySettings(), listLocationsRetry);
>       clientSettingsBuilder.listLocationsSettings().setRetrySettings(listLocationsRetrySettings);
>       if (LOGGER.isTraceEnabled()) {
>         LOGGER.trace("Configured method-level retry settings for listLocations from properties.");
>       }
>     }
>     Retry getLocationRetry = clientProperties.getGetLocationRetry();
>     if (getLocationRetry != null) {
>       RetrySettings getLocationRetrySettings =
>           RetryUtil.updateRetrySettings(
>               clientSettingsBuilder.getLocationSettings().getRetrySettings(), getLocationRetry);
>       clientSettingsBuilder.getLocationSettings().setRetrySettings(getLocationRetrySettings);
>       if (LOGGER.isTraceEnabled()) {
>         LOGGER.trace("Configured method-level retry settings for getLocation from properties.");
>       }
>     }
>     Retry setIamPolicyRetry = clientProperties.getSetIamPolicyRetry();
>     if (setIamPolicyRetry != null) {
>       RetrySettings setIamPolicyRetrySettings =
>           RetryUtil.updateRetrySettings(
>               clientSettingsBuilder.setIamPolicySettings().getRetrySettings(), setIamPolicyRetry);
>       clientSettingsBuilder.setIamPolicySettings().setRetrySettings(setIamPolicyRetrySettings);
>       if (LOGGER.isTraceEnabled()) {
>         LOGGER.trace("Configured method-level retry settings for setIamPolicy from properties.");
>       }
>     }
>     Retry getIamPolicyRetry = clientProperties.getGetIamPolicyRetry();
>     if (getIamPolicyRetry != null) {
>       RetrySettings getIamPolicyRetrySettings =
>           RetryUtil.updateRetrySettings(
>               clientSettingsBuilder.getIamPolicySettings().getRetrySettings(), getIamPolicyRetry);
>       clientSettingsBuilder.getIamPolicySettings().setRetrySettings(getIamPolicyRetrySettings);
>       if (LOGGER.isTraceEnabled()) {
>         LOGGER.trace("Configured method-level retry settings for getIamPolicy from properties.");
>       }
>     }
>     Retry testIamPermissionsRetry = clientProperties.getTestIamPermissionsRetry();
>     if (testIamPermissionsRetry != null) {
>       RetrySettings testIamPermissionsRetrySettings =
>           RetryUtil.updateRetrySettings(
>               clientSettingsBuilder.testIamPermissionsSettings().getRetrySettings(),
>               testIamPermissionsRetry);
>       clientSettingsBuilder
>           .testIamPermissionsSettings()
>           .setRetrySettings(testIamPermissionsRetrySettings);
>       if (LOGGER.isTraceEnabled()) {
>         LOGGER.trace(
>             "Configured method-level retry settings for testIamPermissions from properties.");

```

Reply:
Removing `rest_numeric_enums` causes issues too:
```
diff -r /usr/local/google/home/diegomarquezp/Desktop/spring-cloud-gcp/spring-cloud-generator/scripts/../showcase/showcase-spring-starter-generated/src/main/java/com/google/showcase/v1beta1/spring/TestingSpringAutoConfiguration.java /usr/local/google/home/diegomarquezp/Desktop/spring-cloud-gcp/spring-cloud-generator/scripts/../showcase/showcase-spring-starter/src/main/java/com/google/showcase/v1beta1/spring/TestingSpringAutoConfiguration.java
96a97,99
>     if (this.clientProperties.getUseRest()) {
>       return TestingSettings.defaultHttpJsonTransportProviderBuilder().build();
>     }   
118c121,129
<     TestingSettings.Builder clientSettingsBuilder = TestingSettings.newBuilder();
---
>     TestingSettings.Builder clientSettingsBuilder;
>     if (this.clientProperties.getUseRest()) {
>       clientSettingsBuilder = TestingSettings.newHttpJsonBuilder();
>       if (LOGGER.isTraceEnabled()) {
>         LOGGER.trace("Using REST (HTTP/JSON) transport.");
>       }
>     } else {
>       clientSettingsBuilder = TestingSettings.newBuilder();
>     }   

```

I'll keep checking on the other params

Reply:
`include_samples` and `gapic_yaml` were unnecessary. The other ones produce changes if removed.

Reply:
SG, we can keep the ones produce changes.